src/libs/highlevel/README.md

```diff
@@ -142,7 +142,31 @@ Errors which do not originate inside the high-level API itself are wrapped into
 `ELEKTRA_ERROR_CODE_LOW_LEVEL`. The high-level Error API provides methods (`elektraKDBError*`) to access the properties of the low-level
 error. You can also access the key to which the error was originally attached, as well as any possible low-level warnings.
 
-TODO: explanation of low-level code or removal of this feature for this release?
+To get the original low-level error code, description, severity, group, module and reason you can use these functions:
+```c
+int elektraKDBErrorCode (const ElektraError * error);
+const char * elektraKDBErrorDescription (const ElektraError * error);
+ElektraErrorSeverity elektraKDBErrorSeverity (const ElektraError * error);
+ElektraKDBErrorGroup elektraKDBErrorGroup (const ElektraError * error);
+ElektraKDBErrorModule elektraKDBErrorModule (const ElektraError * error);
+const char * elektraKDBErrorReason (const ElektraError * error);
+```
+
+To iterate over all the warnings use the following to functions:
+
+```c
+int elektraKDBErrorWarningCount (const ElektraError * error);
+ElektraError * elektraKDBErrorGetWarning (const ElektraError * error, int index);
```

Comment:
Do we always use int for index?

Reply:
No, but in this case `int` is more than enough. The generated functions in `kdbError.h` only work for warnings with index 0-99. After that the functions overwrite the warning at index 0. Also the way the warnings array works means that at that point the warning count is also reset.

If we rework our error concept we should change that but for now I don't think it is a problem because generating more than 100 warnings should be very rare.

Reply:
> If we rework our error concept

I would prefer if we now only publish a minimal error API, so that we will not have any compatibility problems when we rework the API. So I would say that we remove the low-level error API for now.

> because generating more than 100 warnings should be very rare.

That might be. We should be consistent with our types nevertheless.